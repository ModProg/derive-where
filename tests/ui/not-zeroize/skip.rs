use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Copy; T)]
struct UnsupportedTrait<T>(#[derive_where(skip(Copy))] PhantomData<T>);

fn main() {}
