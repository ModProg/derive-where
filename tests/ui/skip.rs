use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Debug)]
struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateSkipSeparate<T>(
	#[derive_where(skip)]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);

#[derive_where(Debug)]
struct OverridingSkip<T>(
	#[derive_where(skip)]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct UnderridingSkip<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSeparate<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);

#[derive_where(Clone, Copy)]
struct SkipCloneWhileCopy<T>(#[derive_where(skip(Clone))] PhantomData<T>);

fn main() {}
