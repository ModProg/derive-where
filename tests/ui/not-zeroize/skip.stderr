error: unsupported skip group, expected one of Clone, Debug, EqHashOrd, Hash
 --> tests/ui/not-zeroize/skip.rs:6:48
  |
6 | struct UnsupportedTrait<T>(#[derive_where(skip(Copy))] PhantomData<T>);
  |                                                ^^^^
